app/models/manageiq/providers/ibm_cloud/inventory/parser/object_storage.rb

```diff
@@ -0,0 +1,53 @@
+class ManageIQ::Providers::IbmCloud::Inventory::Parser::ObjectStorage < ManageIQ::Providers::IbmCloud::Inventory::Parser
+  require_nested :StorageManager
+
+  BUCKET_TAB_LIMIT = 1000
+
+  def parse
+    buckets
+  end
+
+  private
+
+  def buckets
+    collector.buckets.each do |bucket|
+      bucket_id = bucket['name']
+
+      total_size, total_count = process_bucket(bucket_id)
+
+      persister.cloud_object_store_containers.build(
+        :ems_ref      => bucket_id,
+        :key          => bucket_id,
+        :bytes        => total_size,
+        :object_count => total_count
+      )
+    end
+  end
+
+  def process_bucket(bucket_id)
+    total_size = total_count = 0
+
+    BUCKET_TAB_LIMIT.times do
+      objects = collector.objects(bucket_id)
+      token = objects[:continuation_token]
```

Comment:
This type of logic should live in the collector since it is related to actually fetching data from the provider API.  You can yield each object to the caller like:
```
collector.rb
def objects(bucket_id)
  params = {}
  1_000.times do
    objects = connection.list_objects_v2({:bucket => bucket_id}, params)
    params = {:continuation_token => objects[:continuation_token]}
    objects[:contents].to_a.each { |content| yield content }

    break if params[:contiuation_token].nil?
  end
end
```

Then your parser is just
```
parser.rb
collector.object_contents(bucket_id) do |content|
  ...
end
```

Reply:
Sure, we can refactor that.

Reply:
I'm working to clean up some of these old TODOs: #318